dbusmock/__main__.py

```diff
@@ -119,8 +128,33 @@ def parse_args():
     if args.template:
         main_object.AddTemplate(args.template, parameters)
 
-    dbusmock.mockobject.objects[args.path] = main_object
+    if args.shell:
+        cmd = [os.environ['SHELL']]
+    else:
+        cmd = args.exec
 
     libglib = ctypes.cdll.LoadLibrary('libglib-2.0.so.0')
-    while should_run:
-        libglib.g_main_context_iteration(None, True)
+
+    dbusmock.mockobject.objects[args.path] = main_object
+
+    if cmd:
+        with subprocess.Popen(cmd) as cmd_proc:
+            @ctypes.CFUNCTYPE(ctypes.c_bool)
+            def check_process():
+                """ Check if the launched process is still alive """
+
+                if cmd_proc.poll() is None:
+                    return True
+                should_run.pop()
+                return False
+
+            libglib.g_idle_add(check_process)
```

Comment:
This is rather "meh" -- would't it be easier to use [unix_signal_add()](https://docs.gtk.org/glib/func.unix_signal_add.html) to listen for SIGCHILD?

Reply:
It would, except that it doesn't support it :(

Reply:
`(process:325649): GLib-CRITICAL **: 23:11:45.387: g_unix_signal_source_new: assertion 'signum == SIGHUP || signum == SIGINT || signum == SIGTERM || signum == SIGUSR1 || signum == SIGUSR2 || signum == SIGWINCH' failed`

Reply:
Ah right, glib does its own thing with watching processes.. https://docs.gtk.org/glib/func.child_watch_add.html ?

Reply:
Using that now, then!